src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +472,30 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        temp_stage = self._session.get_session_stage()
+        stage_path = path
```

Comment:
Nit: better move these two lines to after input validation.

Reply:
Revised